crates/generator/src/generator.rs

```diff
@@ -842,6 +868,20 @@ impl Generator {
                 .collect(),
             debug_log_buf: run_context.debug_log_buf,
         };
+
+        // Record run result
+        {
+            let span = tracing::Span::current();
+            span.record("exit_code", r.exit_code);
+            if r.exit_code != 0 {
+                tracing::error!(error = "non zero exit code")
+            }
+            span.record("execution_cycles", r.cycles.execution);
+            span.record("virtual_cycles", r.cycles.r#virtual);
+            span.record("read_kv_count", r.read_data_hashes.len());
```

Comment:
`r.read_data_hashes` is not kv, please get kv from StateTracker

Reply:
Fixed. 
Rename "[read|write]_kv_count" to "[read|write]_data_count".
Add "touched_key_count" from StateTracker.